www/on/confirm.html.spt

```diff
@@ -1,33 +1,19 @@
-"""This is called as an internal redirect from /on/blah/associate
-"""
-import uuid
-
 from aspen import Response
-from gratipay.models.participant import NeedConfirmation, Participant
 
-website.connect_tokens = {}
+from gratipay.models.account_elsewhere import AccountElsewhere
 
 [--------]
 
 if user.ANON:
-    raise Response(404)
-
-if 'fake' in request.qs:
-    account = next(iter(user.participant.get_accounts_elsewhere().values()))
-elif not getattr(request, 'internally_redirected_from', None):
-    raise Response(404)
-
-# Compute and store a connect_token.
-# ==================================
-# This is a protection against unauthorized users taking over accounts.
-
-connect_token = uuid.uuid4().hex
-connect_key = (user.participant.username, account.platform, account.user_id)
-website.connect_tokens[connect_key] = connect_token
-
+    raise Response(403)
 
-other = Participant.from_username(account.participant.username)
+account_id = request.qs['id']
+account = AccountElsewhere.from_id(account_id)
+token = request.headers.cookie.get('connect_%s' % account_id)
```

Comment:
What are the security implications of using a cookie here instead of the internal redirect? If an attacker stole this cookie what could they do with it?